index.ts

```diff
@@ -31,11 +31,12 @@ async function listFiles(
 ): Promise<Array<TreeResponseObject | ContentsReponseObject>> {
 	const files = await getDirectoryContentViaTreesApi(repoListingConfig);
 
-	if (!files.truncated) {
+	if (files.truncated) {
+		updateStatus('Warning: It’s a large repo and this it take a long while just to download the list of files. You might want to use "git sparse checkout" instead.');
+	} else if (!repoListingConfig.isPrivate) {
```

Comment:
This logic is hard to follow. `isTruncated` and `isPrivate` are unrelated, they shouldn't be part of a single `else if`